shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1702,7 +1702,11 @@ - (CGRect)firstRectForRange:(UITextRange*)range {
       if (@available(iOS 17, *)) {
         startSelectionRect = _selectionRects[i].rect;
       } else {
-        return _selectionRects[i].rect;
+        // The iOS 16 system highlight does not repect the height returned by `firstRectForRange`
+        // API (unlike iOS 17). So we return CGRectZero to hide it (unless if scribble is enabled).
+        // To support scribble's advanced gestures (e.g. insert a space with a vertical bar),
+        // at least 1 character's width is required.
+        return [self isScribbleAvailable] ? _selectionRects[i].rect : CGRectZero;
```

Comment:
Also could you elaborate a bit on what caused the highlight to show at incorrect location? You mentioned the height isn't respected but that shouldn't affect the offset of the rects right?

Reply:
> Also could you elaborate a bit on what caused the highlight to show at incorrect location?

The system highlight shows at the correct location. It should be removed because we already show the highlight in Flutter.

Reply:
https://github.com/flutter/flutter/issues/136802 seems to indicate it's showing at the top left corner?

Reply:
oh i misunderstood you. that is when scribble is not available, where the FlutterTextInputView is positioned at CGRectZero.

Reply:
Thanks for the explanation. I thought we only receive the rects on iPad but it looks like in the framework we always send the character bounding boxes by default as long as the target platform is iOS.